src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectAccessor.cs

```diff
@@ -125,6 +126,7 @@ public async Task OpenProjectForWriteAsync(ConfiguredProject project, Action<Pro
             Requires.NotNull(project, nameof(project));
             Requires.NotNull(project, nameof(action));
 
+            await ((ConfiguredProject2)project).EnsureProjectEvaluatedAsync();
```

Comment:
why isn't this changed needed in EnterWriteLockAsync?

Reply:
It's probably needed there too. Can you elaborate a little on what EnterWriteLockAsync is meant to be used for? I noticed it's not used at the moment. 

EnsureProjectEvaluatedAsync is meant to be used whenever we need an actual writeable evaluation result.